chatsky/llm/wrapper.py

```diff
@@ -0,0 +1,203 @@
+        elif issubclass(message_schema, BaseModel):
+            # Case if the message_schema desribes Message.text structure
+            structured_model = await self.model.with_structured_output(message_schema)
+            result = await structured_model.invoke(history)
+            result = Message(text=str(result.json()))
+
+        if result.annotations:
+            result.annotations["__generated_by_model__"] = self.name
+        else:
+            result.annotations = {"__generated_by_model__": self.name}
+        return result
+
+    async def condition(self, prompt: str, method: BaseMethod):
+        async def process_input(ctx: Context, _: Pipeline) -> bool:
+            condition_history = [await message_to_langchain(Message(prompt), pipeline=_, source="system"),
+                await message_to_langchain(ctx.last_request, pipeline=_, source="human")]
+            result = method(ctx, await self.model.agenerate([condition_history], logprobs=True, top_logprobs=10))
+            return result
+
+        return process_input
+
+
+def llm_response(model_name: str, prompt: str = "", history: int = 5, filter_func: Callable = lambda *args: True, message_schema: Union[None, Type[Message], Type[BaseModel]] = None):
+    """
+    Basic function for receiving LLM responses.
+    :param ctx: Context object. (Assigned automatically)
+    :param pipeline: Pipeline object. (Assigned automatically)
+    :param model_name: Name of the model from the `Pipeline.models` dictionary.
+    :param prompt: Prompt for the model.
+    :param history: Number of messages to keep in history. `-1` for full history.
+    :param filter_func: filter function to filter messages that will go the models context.
+    """
+
+    async def wrapped(ctx: Context, pipeline: Pipeline) -> Message:
+        model = pipeline.models[model_name]
+        if model.system_prompt == "":
+            history_messages = []
+        else:
+            history_messages = [SystemMessage(model.system_prompt)]
+        current_node = ctx.current_node
+        current_misc = current_node.misc if current_node is not None else None
+        if current_misc is not None:
+            # populate history with global and local prompts
+            if "prompt" in current_misc:
+                node_prompt = current_misc["prompt"]
+                history_messages.append(await message_to_langchain(Message(node_prompt), pipeline=pipeline, source="system"))
+            if "global_prompt" in current_misc:
+                global_prompt = current_misc["global_prompt"]
+                history_messages.append(await message_to_langchain(Message(global_prompt), pipeline=pipeline, source="system"))
+            if "local_prompt" in current_misc:
+                local_prompt = current_misc["local_prompt"]
+                history_messages.append(await message_to_langchain(Message(local_prompt), pipeline=pipeline, source="system"))
```

Comment:
Why this order?
Shouldn't it be "global->local->node"?

Reply:
As far as I recall we've decided to put the "node" prompt on the top as the most important one